File: clients/php/src/FileDBException.php
<?php

declare(strict_types=1);

namespace FileDBv2;

/**
 * Base class for all errors surfaced by the FileDB client.
 *
 * Extends \RuntimeException so existing `catch (\RuntimeException $e)` blocks
 * keep working; catch the more specific NotFoundException / AlreadyExistsException
 * subclasses to handle keyed-CRUD outcomes idiomatically.
 */
class FileDBException extends \RuntimeException
{
}

File: clients/java/src/main/java/com/srjn45/filedbv2/FileDBException.java
package com.srjn45.filedbv2;

/**
 * Base class for FileDB client errors surfaced from engine gRPC status codes.
 *
 * <p>The keyed-CRUD surface (N1) maps the common failure codes onto typed
 * subclasses — {@link NotFoundException} for {@code NOT_FOUND} and
 * {@link AlreadyExistsException} for {@code ALREADY_EXISTS}. Any other status
 * code propagates unchanged as the original {@link io.grpc.StatusRuntimeException}.
 */
public class FileDBException extends RuntimeException {

    public FileDBException(String message) {
        super(message);
    }

    public FileDBException(String message, Throwable cause) {
        super(message, cause);
    }
}
